database/migrations/001-punishments-schema.sql

```diff
@@ -0,0 +1,46 @@
+--------------------------------------------------------------------------------
+-- Up
+--------------------------------------------------------------------------------
+
+CREATE TABLE IF NOT EXISTS punishments (
+	punishType TEXT NOT NULL,
+	userid TEXT NOT NULL,
+	ips TEXT NOT NULL,
+	userids TEXT NOT NULL,
+	expireTime INTEGER NOT NULL,
+	reason TEXT NOT NULL,
+	PRIMARY KEY (punishType, userid)
+);
+
+CREATE TABLE IF NOT EXISTS room_punishments (
+	punishType TEXT NOT NULL,
+	-- roomid:userid
+	id TEXT NOT NULL,
+	ips TEXT NOT NULL,
+	userids TEXT NOT NULL,
+	expireTime INTEGER NOT NULL,
+	reason TEXT NOT NULL,
+	PRIMARY KEY (punishType, id)
+);
+
+CREATE TABLE IF NOT EXISTS shared_ips (
+	ip TEXT NOT NULL,
+	-- type is always "SHARED", but we will store it anyway in case there are more types in the future
```

Comment:
Just leave it out - if more types are added in the future, old rows can be migrated to add "SHARED" in